### DISCUSSION TITLE
👽 Block removal mutation Survived - for method with return type

### DISCUSSION THREAD

**Discussion by Burgyn** [stryker-mutator/stryker-net#2889] [General]
Hi.

I have question. 
I have methods like:

```csharp
public ValueTask ServeNotModifiedAsync(ETagCacheContext context, CancellationToken cancellationToken)
{
    return ValueTask.CompletedTask;
}
```

Stryker tells me "👽 Block removal mutation Survived".
![image](https://github.com/stryker-mutator/stryker-net/assets/5930822/d4d68410-2bb5-4360-b448-58d6dcf76f87)

but I don't understand why. Because if I remove the `return` and leave an empty block, it's compilation error.
Can you advise me what I'm doing wrong?

Thanks a lot.

**Comment by rouke-broersma** [comment]
When we cannot place any other method in a scope, we remove the whole block as a crude last resort. Of course that doesn't compile, so instead of actually removing we replace it with return default;

See: https://github.com/stryker-mutator/stryker-net/discussions/2296

We are still debating how we could best show this to the user, since the intent is not to add the return default, the intent is to remove the block. Currently we don't show this at all, and we agree that this is confusing.

**Comment by dupdob** [comment]
...and in this case, it turns out that `default` is `ValueTask.CompletedTask`, thus Stryker ends with a semantically equivalent mutation.
The best work around for now is to disable block mutations via a Stryker comment:

```csharp
// Stryker disable once Block: results in an equivalent mutation
public ValueTask ServeNotModifiedAsync(ETagCacheContext context, CancellationToken cancellationToken)
{
    return ValueTask.CompletedTask;
}
```

**Comment by Burgyn** [comment]
Thanks a lot 🙏
